lib/fog/storage/google_json/utils.rb

```diff
@@ -19,10 +21,7 @@ def url(params, expires)
 
         def host_path_query(params, expires)
           params[:headers]["Date"] = expires.to_i
-          # implementation from CGI.escape, but without ' ' to  '+' conversion
-          params[:path] = params[:path].b.gsub(/([^a-zA-Z0-9_.\-~]+)/) { |m|
-            '%' + m.unpack('H2' * m.bytesize).join('%').upcase
-          }.gsub("%2F", "/")
+          params[:path] = ::Addressable::URI.encode_component(params[:path], ::Addressable::URI::CharacterClasses::PATH)
```

Comment:
This sure is a lot nicer to read, independent of getting our desired `+` behavior!

(If no one ever responds to your Github issue... I wonder if it's worth opening a PR on `fog-google`? I don't really know the etiquette rules!)